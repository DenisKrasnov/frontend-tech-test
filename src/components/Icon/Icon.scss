.Icon {
  display: block;
}

.blue {
  fill: $icon-color;
}

.white {
  fill: #fff;
}
