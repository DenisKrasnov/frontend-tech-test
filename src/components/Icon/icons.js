const pencil = require("./svg/pencil-sm.svg");
const trash = require("./svg/trash-sm.svg");
const x = require("./svg/x-thin-sm.svg");

export default {
  pencil,
  trash,
  x,
};
