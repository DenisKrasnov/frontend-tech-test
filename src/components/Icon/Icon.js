import React from "react";
import { string, func, oneOf } from "prop-types";
import cn from "classnames";

import icons from "./icons";
import styles from "./Icon.scss";

const Icon = props => {
  const { glyph, onClick, color } = props;
  const Svg = icons[glyph];
  const className = cn(styles.Icon, styles[color]);
  return <Svg className={className} focusable="false" onClick={onClick} />;
};

Icon.defaultProps = {
  onClick: () => {},
  color: "blue",
};

Icon.propTypes = {
  glyph: string.isRequired,
  onClick: func,
  color: oneOf(["white", "blue"]),
};

export default Icon;
