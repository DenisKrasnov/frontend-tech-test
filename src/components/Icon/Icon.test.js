import React from "react";
import { shallow } from "enzyme";

import Icon from "./Icon";

describe("Icon", () => {
  it("renders", () => {
    const component = shallow(<Icon glyph="x" color="white" />);
    expect(component).toMatchSnapshot();
  });

  it("triggers onClick", () => {
    const onClickSpy = jest.fn();
    const component = shallow(
      <Icon glyph="x" color="white" onClick={onClickSpy} />
    );

    component.find(".Icon").simulate("click");
    expect(onClickSpy).toHaveBeenCalled();
  });
});
