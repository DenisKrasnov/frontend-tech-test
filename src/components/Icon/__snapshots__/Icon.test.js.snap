// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders 1`] = `
<test-file-stub
  className="Icon white"
  focusable="false"
/>
`;
